runeq/resources/project.py

```diff
@@ -9,6 +9,311 @@
+                    APPLEWATCH_DYSKINESIA, APPLEWATCH_HEART_RATE,
+                    PERCEPT_TREND_LOG_LFP
+            value:
+            time_interval: Period over which the metric was calculated.
+                Possible time intervals include:
+                    FOURTEEN_DAYS, NINETY_DAYS, PROJECT_ALL
+            created_at: Time patient was added to the cohort (unix timestamp)
+            updated_at: Time cohort patient was last updated (unix timestamp)
+
+        """
+        self.type = type
+        self.data_type = data_type
+        self.time_interval = time_interval
+        self.created_at = created_at
+        self.updated_at = updated_at
+        self.value = value
+
+        super().__init__(
+            id=id,
+            type=type,
+            data_type=data_type,
+            value=value,
+            time_interval=time_interval,
+            created_at=created_at,
+            updated_at=updated_at,
+        )
+
+
+class MetricSet(ItemSet):
+    """
+    A collection of Metrics.
+
+    """
+
+    def __init__(self, items: Iterable[Metric] = ()):
+        """
+        Initialize with Metrics.
+
+        """
+        super().__init__(items=items)
+
+    @property
+    def _item_class(self) -> Type[ItemBase]:
+        """
+        Instance type of items in this set.
+
+        """
+        return Metric
+
+
+class CohortPatient(ItemBase):
+    """
+    A patient who has been placed within a cohort and is now
+    associated with the cohort.
+
+    """
+    def __init__(
+        self,
+        id: str,
+        updated_at: float,
+        created_at: float,
+        created_by: str,
+        updated_by: str,
+        metrics: MetricSet,
+        **attributes
+    ):
+        """
+        Initialize with data.
+
+        Args:
+            id: Unique identifier of the patient
+            created_at: Time patient was added to the cohort (unix timestamp)
+            created_by: Display name of who added the patient to the cohort
+            updated_at: Time cohort patient was last updated (unix timestamp)
+            updated_by: Display name of who updated the cohort patient record
+            **attributes: Other attributes associated with the cohort
+
+        """
+        self.created_at = created_at
+        self.updated_at = updated_at
+        self.created_by = created_by
+        self.updated_by = updated_by
+        self.metrics = metrics
+
+        super().__init__(
+            id=id,
+            created_at=created_at,
+            updated_at=updated_at,
+            created_by=created_by,
+            updated_by=updated_by,
+            metrics=metrics,
+            **attributes,
+        )
+
+    def to_dict(self) -> dict:
+        """
+        Dictionary representation of the Cohort Patient attributes.
+
+        """
+        attrs = self._attributes.copy()
+        attrs["metrics"] = self.metrics.to_list()
+        return attrs
+
+
+# TODO: Determine if project_id should be part of class
```

Comment:
Are the TODOs still to be answered?

Reply:
Nope! This is still a question. @abhinavpottabathula @anyu do you all think project ID should be a part of this class? I couldn't decide because the project id is used to query the patients, so the user already has the project id...

Reply:
Yea, I don't see the need!